doc/groups.md

```diff
@@ -1 +1,519 @@
+404 | Not Found | Groups not found in database  
+500 | Internal Server Error | some un-handle error in server 
+200 | OK | OK
+```
+{
+    "status_code": "CDC-403",
+    "status_message": "Groups is not owned by user login",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "Data Changed",
+    "data": {
+        "id": "1",
+        "title": "Admin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-01 00:00:00"
+    }
+} 
+```
+### Logic
+#### Endpoint Validation
+- must be add parameter id
+- id must exist in database
+- must owned by user login
+
+#### Validation
+No Validation
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Groups is not owned by user login
+
+Response HTTP Status Code : 403
+
+Response Payload :
+```
+{
+    "status_code": "CDC-403",
+    "status_message": "Groups is not owned by user login",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+- param id is not exist in database
+
+Endpoint : /groups/100
+
+Response HTTP Status Code : 404
+
+Response Payload :
+```
+{
+    "status_code": "CDC-404",
+    "status_message": "data not found",
+    "data": null
+}
+```
+
+#### Case : Positive Case 1
+
+- param id is exist in database
+
+Endpoint : /groups/1
+
+Headers :
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+Email | administrator@gmail.com
+
+Request Payload :
+```
+{
+   "title": "myAdmin",
+   "description": "My Administrator"
+}
+```
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "Data Changed",
+    "data": {
+        "id": "1",
+        "title": "myAdmin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-11 12:00:00"
+    }
+}
+```
+
+#### Case : Positive Case 2
+
+Request Payload : empty
+
+Endpoint : /groups/1
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "No data was changed",
+    "data": {
+        "id": "1",
+        "title": "myAdmin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-11 12:00:00"
+    }
+}
+```
+
+#### Case : Positive Case 3
+
+Request Payload :
+```
+{}
+```
+Endpoint : /groups/1
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "No data was changed",
+    "data": {
+        "id": "1",
+        "title": "myAdmin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-11 12:00:00"
+    }
+}
+```
+
+## <a name="view"></a>View All Groups
+
+### Endpoint
+GET /groups
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+Email | administrator@mail.com
+
+### Request Payloads
```

Comment:
### Param Query
Name | Example Value | Description
--- | --- | ---
search | admin | keyword for search groups. default is empty.
page | 1 | current page. default 1.
limit | 20 | limit data in 1 pages. default 20.
order | groups.title | order of list. default groups.id
sort | asc | sort of list. option value is asc|desc. default value is desc

Example URL : GET /groups?search=admin&order=groups.title

Reply:
```
### Param Query
Name | Example Value | Description
--- | --- | ---
search | admin | keyword for search groups. default is empty.
page | 1 | current page. default 1.
limit | 20 | limit data in 1 pages. default 20.
order | groups.title | order of list. default groups.id
sort | asc | sort of list. option value is asc|desc. default value is desc

Example URL : GET /groups?search=admin&order=groups.title
```

Reply:
there is no request payload, please change with param query